File: Items/kusa_Split and implode from bank.py
# @description kusa_Split and implode from bank
# @version 1.0
# @author Kusa
# @website https://thomashugofritz.wixsite.com/website
# @donation https://paypal.me/tfkusa?country.x=FR&locale.x=fr_FR

import subprocess
import reapy

def run_external_program():
    # Use reapy to get REAPER's resource path
    reaper_resource_path = reapy.reascript_api.GetResourcePath()

    # Specify the relative path to the standalone script within the 'Python' subfolder
    relative_script_path = "/Scripts/kusa_reascriptsWithDependencies/lib/kusa_banktotakes.py"

    # Combine them to form the full path to the script
    script_path = reaper_resource_path + relative_script_path

    print(script_path)

    # Properly quote the path
    quoted_script_path = f"'{script_path}'"

    # Construct AppleScript command to run the script in Terminal
    applescript_command = f"""
    tell application "Terminal"
        do script "python3 {quoted_script_path}"
    end tell
    """
    subprocess.run(["osascript", "-e", applescript_command])

run_external_program()

File: Items/kusa_Split and implode from bank.lua
-- @description kusa_Split and implode from bank
-- @version 1.01
-- @author Kusa
-- @website https://thomashugofritz.wixsite.com/website
-- @donation https://paypal.me/tfkusa?country.x=FR&locale.x=fr_FR

function runExternalProgram()
    local reaperResourcePath = reaper.GetResourcePath()

    local relativeScriptPath = "/Scripts/kusa_reascriptsWithDependencies/lib/kusa_banktotakes.py"
    if reaper.GetOS() == "Win" then
        relativeScriptPath = "\\Scripts\\kusa_reascriptsWithDependencies\\lib\\kusa_banktotakes.py"
    end

    local scriptPath = reaperResourcePath .. relativeScriptPath

    local command
    if reaper.GetOS() == "Win" then
        command = "python \"" .. scriptPath .. "\""
    else
        local appleScriptCommand = "tell application \"Terminal\" to do script \"python3 '" .. scriptPath .. "'\""
        command = "osascript -e " .. string.format("%q", appleScriptCommand)
    end

    os.execute(command)
end

runExternalProgram()
